plugin/trino-hive/src/main/java/io/trino/plugin/hive/AbstractHiveAcidWriters.java

```diff
@@ -52,7 +52,7 @@
     // The original file path looks like .../nnnnnnn_m(_copy_ccc)?
     public static final Pattern ORIGINAL_FILE_PATH_MATCHER = Pattern.compile("(?s)(?<rootDir>.*)/(?<filename>(?<bucketNumber>[\\d]+)_(?<rest>.*)?)$");
     // After compaction, the bucketPath looks like .../base_nnnnnnn(_vmmmmmmm)?/bucket_bbbbb(_aaaa)?
-    public static final Pattern BASE_PATH_MATCHER = Pattern.compile("(?s)(?<rootDir>.*)/(?<dirStart>base_[\\d]+(_v[\\d]+)?)/(?<filenameBase>bucket_(?<bucketNumber>[\\d]+))(?<attemptId>_[\\d]+)?$");
+    public static final Pattern BASE_PATH_MATCHER = Pattern.compile("(?s)(?<rootDir>.*)/(?<dirStart>base_[-]?[\\d]+(_v[\\d]+)?)/(?<filenameBase>bucket_(?<bucketNumber>[\\d]+))(?<attemptId>_[\\d]+)?$");
```

Comment:
Is the problem about deletes only?
this BASE_PATH_MATCHER is used for selects too, right?

Reply:
I didn't check that before because I assumed it is only used in the Writer. I checked and in fact it is. 
Selects work fine and don't touch this Pattern at all.

Reply:
right, have misread